File: pset2/readibility/readibility.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <stdlib.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);
int main(void)
{
    // prompt for input
    string text = get_string("Text: ");

    // count letters
    int letter_count = count_letters(text);
    printf("letter count: %d\n", letter_count);

    // count words
    int word_count = count_words(text);
    printf("word count: %d\n", word_count);

    // count sentences
    int sentence_count = count_sentences(text);
    printf("sentence count: %d\n", sentence_count);

    // index = 0.0588 * L - 0.296 * S - 15.8

    float L = letter_count / (float)word_count * 100;
    float S = sentence_count / (float)word_count * 100;

    printf("L is: %f\nS is: %f\n", L, S);
    float i = 0.0588 * L - 0.296 * S - 15.8;
    int index = round(i);

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %d\n", index);
    }
}

int count_letters(string text)
{
    int count = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        // only count alphabets as letters
        if (isalpha(text[i]))
        {
            count++;
        }
    }
    return count;
}

int count_words(string text)
{
    int count = 1;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        // spaces separate words
        if (isspace(text[i]))
        {
            count++;
        }
    }
    return count;
}

int count_sentences(string text)
{
    int count = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        // for this problem, a sentence ends with . ! ? only
        if ((int)text[i] == (int)'.' || (int)text[i] == (int)'!' || (int)text[i] == (int)'?')
        {
            count++;
        }
    }
    return count;
}

File: pset6/sentimental-readibility/readibility.py
from cs50 import get_string


def main():
    # Prompt for user input
    text = get_string("Text: ")

    # count letters
    letter_count = count_letters(text)
    print("letter count:", letter_count)

    # count words
    word_count = count_words(text)
    print(f"word count: {word_count}")

    # count sentences
    sentence_count = count_sentences(text)
    print(f"sentence count: {sentence_count}")

    # calculate Coleman-Liau index
    L = letter_count / word_count * 100
    S = sentence_count / word_count * 100

    i = 0.0588 * L - 0.296 * S - 15.8
    index = round(i)

    if index < 1:
        print("Before Grade 1")
    elif index > 10:
        print("Grade 16+")
    else:
        print(f"Grade {index}")


def count_letters(str):
    count = 0
    for c in str:
        if c.isalpha():
            count += 1
    return count


def count_words(text):
    count = 1
    for c in text:
        if c.isspace():
            count += 1
    return count


def count_sentences(text):
    count = 0
    for c in text:
        if c == '.' or c == '!' or c == '?':
            count += 1
    return count


main()
